### PR TITLE
Bug 1750825: consistent behaviour in top-consumer graph

### PR DESCRIPTION
**Pull request by a2batic**
Top Consumers chart changes colors every few seconds, PR fixes
the problem
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1750825
Co-authored-by: Afreen Rahman afrahman@redhat.com
Signed-off-by: Kanika <kmurarka@redhat.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/utils.ts [hunk 1]
```diff
@@ -19,6 +19,25 @@ export const sortResources: SortResourcesProps = (a, b) => {
...
   const y = _.get(b, ['values', bVal.length - 1, 1]);
+  if (x === y) {
+    const aNameVal = _.get(a, 'metric');
+    const bNameVal = _.get(b, 'metric');
+    if (aNameVal.storageclass) {
+      return aNameVal.storageclass < bNameVal.storageclass
+        ? -1
+        : aNameVal.storageclass > bNameVal.storageclass
+        ? 1
+        : 0;
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/utils.ts]: Looks repetitive. create a function and pass` aNameVal, BNameVal and metricType`

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/utils.ts [hunk 2]
```diff
@@ -19,6 +19,25 @@ export const sortResources: SortResourcesProps = (a, b) => {
   const bVal = _.get(b, 'values');
   const x = _.get(a, ['values', aVal.length - 1, 1]);
   const y = _.get(b, ['values', bVal.length - 1, 1]);
+  if (x === y) {
+    const aNameVal = _.get(a, 'metric');
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/utils.ts]: We have metric type available with us[1].

Pass metric type and use it.

```suggestion
topConsumerStatsResult.sort(sortResources(metricType))
```
[1]. https://github.com/openshift/console/blob/master/frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card-body.tsx#L48

Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/utils.ts]: @cloudbehl passing any additional argument to compare function inside `Array.prototype.sort()` can lead to a complicated code. I have passed it as in a normal function.

**Comment by a2batic** [comment]
@cloudbehl addressed review comments, please review.

**Comment by a2batic** [comment]
Closing this PR as bug is not a blocker bug.

**Pull request closed without merging**